code/datums/beam.dm

```diff
@@ -41,9 +44,10 @@
 		return
 	recalculating = TRUE
 	timing_id = null
-	if(origin && target && get_dist(origin,target)<max_distance && origin.z == target.z)
-		var/origin_turf = get_turf(origin)
-		var/target_turf = get_turf(target)
+	var/turf/origin_turf = get_turf(origin)
+	var/turf/target_turf = get_turf(target)
+	curr_distance = get_dist(origin_turf, target_turf)
+	if(origin && target && curr_distance < max_distance && origin_turf.z == target_turf.z)
```

Comment:
```suggestion
	if(origin_turf && target_turf && curr_distance < max_distance && origin_turf.z == target_turf.z)
```

Checking origin and target isn't enough, could runtime if it reaches origin_turf.z bit and origin doesn't have a turf.

```suggestion
	if(!isnull(curr_distance) && curr_distance < max_distance && origin_turf.z == target_turf.z)
```

Alternatively do this, but you'll need to make curr_distance initialize as null and clear it in appropriate spots since you reuse it multiple times throughout.